ledger-core/src/core/wallet/WalletStore.cpp

```diff
@@ -19,10 +19,34 @@ namespace ledger {
             return Option<api::Currency>();
         }
 
+        void WalletStore::getCurrency(
+            const std::string & name,
+            const std::function<void(std::experimental::optional<api::Currency>, std::experimental::optional<api::Error>)> & callback
+        ) {
+            auto self = shared_from_this();
+            Future<api::Currency>::async(_services->getDispatcher()->getMainExecutionContext(), [self, name] () {
```

Comment:
Maybe we can use in-place capture for `self` variable to avoid a copy and "under the hood" count references involved in `shared_ptr` ?

Reply:
You mean a _move_? Can lambda support that?

Reply:
I mean in-place capture variable declaration (waouh such a long term) such as :
`[self = shared_from_this(), ...](...) {}`

Reply:
Hm. I didn’t know you could do that. Are you sure it’s not going to make an extra copy as well?

Reply:
It's quite new in lambda capture; the `self` variable will be directly declared as lambda variable instead of captured variable from the stack context.

Reply:
Actually, not here, because the value is used below in the same scope. I’ll apply the idea to the rest of the file whenever possible though.